src/cmd/run.go

```diff
@@ -63,13 +67,61 @@ var runCmd = &cobra.Command{
 			message.Fatalf(err, "Cannot unmarshal %s", config.TaskFileLocation)
 		}
 
-		if config.ListTasks {
+		if config.ListTasks || config.ListAllTasks {
 			rows := [][]string{
 				{"Name", "Description"},
 			}
 			for _, task := range tasksFile.Tasks {
 				rows = append(rows, []string{task.Name, task.Description})
 			}
+			// If ListAllTasks, add tasks from included files
+			if config.ListAllTasks {
+				var includedTasksFile types.TasksFile
+				var fullPath string
+				templatePattern := `\${[^}]+}`
+				re := regexp.MustCompile(templatePattern)
+				for _, include := range tasksFile.Includes {
+					// get included TasksFile
+					for includeName, includeFileLocation := range include {
+						// check for templated variables in includeFileLocation value
+						if re.MatchString(includeFileLocation) {
+							runner := runner.Runner{
+								TemplateMap: map[string]*zarfUtils.TextTemplate{},
+								TasksFile:   tasksFile,
+								TaskNameMap: map[string]bool{},
+							}
+							runner.PopulateTemplateMap(runner.TasksFile.Variables, config.SetRunnerVariables)
+							includeFileLocation = runner.TemplateString(includeFileLocation)
```

Comment:
I think as part of this we can also move the Actions code into a different file, that way this entire codebase isn't just in `runner.go` 😆

Reply:
> Got an idea for a refactor: instead of having to instantiate a runner, let's just have TemplateString take in the TemplateMap as an arg, and make TemplateString a standalone fn

Yeah, didn't love creating this "filler" runner, in order to get rid of it though, we would also need to refactor the PopulateTemplateMap function to just return the template map that TemplateString would need, but then this snowballs a bit, because everywhere TemplateString gets called in the runner.go file (importTasks, loadIncludedTaskFile, placeFiles...) we will need to pass in this templateMap... you end up having to pass that map around everywhere vs being able to just grab it from the runner, which is what is currently happening.

Reply:
workaround found by @UncleGedd to limit concern from above